lib/spack/spack/bootstrap.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import contextlib
+import os
+import sys
+
+import llnl.util.filesystem as fs
+import llnl.util.tty as tty
+
+import spack.architecture
+import spack.config
+import spack.paths
+import spack.repo
+import spack.spec
+import spack.store
+import spack.user_environment as uenv
+import spack.util.executable
+from spack.util.environment import EnvironmentModifications
+
+
+@contextlib.contextmanager
+def system_python_context():
```

Comment:
Yes, I proposed `current_python_interpreter_context` for lack of a better name. The main point is that this may not be something coming from the OS package manager (since we use "system" in other contexts to indicate that).

Reply:
@becker33 Maybe:
```python
with spack_python_interpreter():
    ...
```
?

Reply:
Let me know if 2ac6544 is fine